lib/ReactGridLayout.jsx

```diff
@@ -250,10 +250,18 @@ export default class ReactGridLayout extends React.Component {
 
     // Set state
     this.setState({
-      layout: compact(layout, this.props.verticalCompact),
       activeDrag: null,
       oldDragItem: null
     });
+
+    // Set State and callback, but only if layout changed
+    const newLayout = compact(layout, this.props.verticalCompact);
+    if (!isEqual(newLayout, this.state.layout)) {
```

Comment:
@STRML your are absolutely right about that. Coming from Immutable States I always tend to make those mistakes where in place changes are made. Sorry for the inconvenience.

I already reverted this: 

https://github.com/menelike/react-grid-layout/commit/c5e6d8cb75bb82a8c0fc2aab06336e55e1eb2ae1

What would the proper way here, another PR? I do not want to pollute your PRs.